browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,461 @@
+    data_resource = resource_bundle.LoadDataResourceString(id);
+  } else {
+    data_resource = resource_bundle.GetRawDataResource(id).as_string();
+  }
+
+  return data_resource;
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+    const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources = g_brave_browser_process->
+      ad_block_service()->UrlCosmeticResources(url);
+
+  if (!resources || !resources->is_dict()) {
+    return result_list;
+  }
+
+  base::Optional<base::Value> regional_resources = g_brave_browser_process->
+      ad_block_regional_service_manager()->UrlCosmeticResources(url);
+
+  if (regional_resources && regional_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*regional_resources), &*resources, /*force_hide=*/false);
+  }
+
+  base::Optional<base::Value> custom_resources = g_brave_browser_process->
+      ad_block_custom_filters_service()->UrlCosmeticResources(url);
+
+  if (custom_resources && custom_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*custom_resources), &*resources, /*force_hide=*/true);
+  }
+
+  result_list->Append(std::move(*resources));
+
+  return result_list;
+}
+
+std::unique_ptr<base::ListValue> GetHiddenClassIdSelectorsOnTaskRunner(
+    const std::vector<std::string>& classes,
+    const std::vector<std::string>& ids,
+    const std::vector<std::string>& exceptions) {
+  base::Optional<base::Value> hide_selectors = g_brave_browser_process->
+      ad_block_service()->HiddenClassIdSelectors(classes, ids, exceptions);
+
+  base::Optional<base::Value> regional_selectors = g_brave_browser_process->
+      ad_block_regional_service_manager()->
+          HiddenClassIdSelectors(classes, ids, exceptions);
+
+  base::Optional<base::Value> custom_selectors = g_brave_browser_process->
+      ad_block_custom_filters_service()->
+          HiddenClassIdSelectors(classes, ids, exceptions);
+
+  if (hide_selectors && hide_selectors->is_list()) {
+    if (regional_selectors && regional_selectors->is_list()) {
+      for (auto i = regional_selectors->GetList().begin();
+          i < regional_selectors->GetList().end(); i++) {
+        hide_selectors->Append(std::move(*i));
+      }
+    }
+  } else {
+    hide_selectors = std::move(regional_selectors);
+  }
+
+  auto result_list = std::make_unique<base::ListValue>();
+  if (hide_selectors && hide_selectors->is_list()) {
+    result_list->Append(std::move(*hide_selectors));
+  }
+  if (custom_selectors && custom_selectors->is_list()) {
+    result_list->Append(std::move(*custom_selectors));
+  }
+
+  return result_list;
+}
+
+bool IsVettedSearchEngine(const std::string& host) {
+  for (size_t i = 0;
+      i < BraveCosmeticResourcesTabHelper::vetted_search_engines_.size();
+      i++) {
+    size_t found_pos = host.find(
+        BraveCosmeticResourcesTabHelper::vetted_search_engines_[i]);
```

Comment:
would this match for hosts like `notyahoo.com` ? and also sites like http://onesearch.net/ which may not be owned by the search engine.

Reply:
This check is used as an optimization to minimize page re-layout on commonly visited pages with only first-party sponsored content, so no major harm in the second case.

Although, it is true that the list ought to be revisited occasionally (cc @pes10k)

Reply:
Not it will not match for `notyahoo.com`, we are good here. We will probably have to create a follow up if we want to optimize a second part, but as @antonok-edm mentioned not a major harm.